sympy/algebras/tests/test_quaternion.py

```diff
@@ -57,6 +57,12 @@ def test_quaternion_complex_real_addition():
     assert q1 - q1 == q0
 
 
+def test_quaternion_evalf():
+    assert Quaternion(sqrt(2), 0, 0, sqrt(3)).evalf() == Quaternion(sqrt(2).evalf(), 0, 0, sqrt(3).evalf())
+    assert Quaternion(sqrt(2)*I, 0, 0, sqrt(3)+sqrt(4)*I).evalf() == \
+    Quaternion(sqrt(2).evalf()*I, 0, 0, sqrt(3).evalf() + sqrt(4).evalf()*I)
```

Comment:
Only the first test is needed. I don't think that using complex numbers in the arguments of a quaternion should be allowed...

Reply:
But in the docs of Quaternion class it was mentioned we can define quaternions over complex fields -
    >>> from sympy import symbols, I
    >>> x = symbols('x')
    >>> q1 = Quaternion(x, x**3, x, x**2, real_field = False)
    >>> q2 = Quaternion(3 + 4*I, 2 + 5*I, 0, 7 + 8*I, real_field = False)
    >>> q1
    x + x**3*i + x*j + x**2*k
    >>> q2
    (3 + 4*I) + (2 + 5*I)*i + 0*j + (7 + 8*I)*k
so i guess we can include that particular test case.

Reply:
I would prefer not to add new tests involving complex numbers

Reply:
@oscarbenjamin I've removed the test involving complex numbers

Reply:
Can you also add the test mentioned in the issue?
`Quaternion(1/sqrt(2), 0, 0, 1/sqrt(2)).evalf()`

This PR is ready after that.